tests/e2e/support/api/keycloak/ocisUserToken.ts

```diff
@@ -0,0 +1,141 @@
+import fetch from 'node-fetch'
+import { TokenEnvironmentFactory } from '../../environment'
+import { config } from '../../../config'
+import { User } from '../../types'
+
+interface ocisTokenForKeycloak {
+  access_token: string
+  refresh_token: string
+}
+
+const authorizationEndpoint = config.keycloakUrl + '/realms/oCIS/protocol/openid-connect/auth'
+const tokenEndpoint = config.keycloakUrl + '/realms/oCIS/protocol/openid-connect/token'
+const redirectUrl = config.backendUrl + '/oidc-callback.html'
+
+async function getAuthorizationEndPoint() {
+  const loginParams = {
+    client_id: 'web',
+    redirect_uri: redirectUrl,
+    response_mode: 'query',
+    response_type: 'code',
+    scope: 'openid profile email'
+  }
+  const queryString = new URLSearchParams(loginParams).toString()
+  const authorizationUrl = `${authorizationEndpoint}?${queryString}`
+
+  const authorizationResponse = await fetch(authorizationUrl, {
+    method: 'GET',
+    redirect: 'manual'
+  })
+
+  if (authorizationResponse.status === 302) {
+    const locationHeader = authorizationResponse.headers.get('location')
+    const urlParams = new URLSearchParams(new URL(locationHeader).search)
+    const errorDescription = urlParams.get('error_description')
+    throw new Error(`Unexpected redirection. ${errorDescription}`)
+  } else if (authorizationResponse.status !== 200) {
+    throw new Error(
+      `Authorization failed: Expected status code to be 200 but received ${authorizationResponse.status}. \nMessage: ${authorizationResponse.statusText}`
+    )
+  }
+
+  const cookies = authorizationResponse.headers.raw()['set-cookie']?.[0]
+  const htmlData = await authorizationResponse.text()
+
+  // authorization url for login is send back from server in the HTML body.
+  const match = htmlData.match(/action="([^"]+)"/)
+  if (!match) {
+    throw new Error('No authorization url found in the HTML response body.')
+  }
+  const auhorizationUrl = match[1]
+  return [auhorizationUrl, cookies]
```

Comment:
```suggestion
  return [match[1], cookies]
```

Reply:
In the future, even I would be confused because I would have no idea what `match[1]`  even mean.  The variable `auhorizationUrl makes it easier to understand what value it holds.